pyfaidx/__init__.py

```diff
@@ -607,7 +607,7 @@ def build_index(self):
     def write_fai(self):
         with self.lock:
             with open(self.indexname, 'w') as outfile:
-                for line in self._index_as_string:
+                for line in self._index_as_string():
```

Comment:
Good catch. This is bizarre. I didn't know that you could call a generator without parenthesis, and this is surely what I meant to do in the first place.